### PR TITLE
Add basic support for Broadcom Tomahawk2 and Tomahawk3

### PR DESCRIPTION
**Pull request by pudelkoM**
This PR updates OpenNSA to version 6.5.19.1 and adds basic support for Tomahawk2/3 chips.
Stratum should be able to start and program the dataplane, but some features like PacketIO will not work and require chip-specific augmentations.

TODOs:
- [x] Find missing configuration values for TH2/TH3
- [x] Confirm that the version update did not break anything
- [ ] ~Add some new ChassisConfigs?~
- [x] Add new platform types to oc converter

Thanks @habeebmohammed for providing the initial code changes.

### CODE REVIEW

File: stratum/hal/lib/bcm/bcm_chassis_manager.cc [hunk 1]
```diff
@@ -909,6 +915,7 @@ ::util::Status BcmChassisManager::GenerateBcmChassisMapFromConfig(
   }
 
   // Check for max num of ports per chip.
+  // TODO(max): find max port number fot TH2/TH3
```
Comment (pudelkoM) [inline stratum/hal/lib/bcm/bcm_chassis_manager.cc:918]: Maybe there are some references in the SDK or spec sheets?

**Comment by pudelkoM** [comment]
@habeebmohammed if you have chassis configs for any new devices with these chips, feel free to add them here.

**Comment by pudelkoM** [comment]
@abhilashendurthi it would be good to get a green light from the CC program to confirm that this update still works.

**Comment by abhilashendurthi** [comment]
@pudelkoM, just to confirm, you would like to see if the existing switches still work with opennsa upgrade to 6.5.19.1?
We currently have AS7712, AS6712 running with bcm_opennsa for CCP. I can run tests AS7712, AS6712 to verify if the changes are still working.

**Comment by pudelkoM** [comment]
> @pudelkoM, just to confirm, would you like to see if the existing switches still work with opennsa upgrade to 6.5.19.1?
> We currently have AS7712, AS6712 running with bcm_opennsa for CCP. I can run tests AS7712, AS6712 to verify if the changes are still working.

Yes, exactly that. It seems to start just fine in my tests, but it would be nice to have more thorough checks as well.

**Comment by abhilashendurthi** [comment]
sounds good, I will test and update the outcome.

**Comment by abhilashendurthi** [comment]
Tested on `AS6712` with opennsa debian package and `AS6712` with both opennsa debian package and opennsa docker image. All TestVectors run successfully as expected.

**Review by bocon13 — APPROVED** [review]
seems reasonable

**Comment by pudelkoM** [comment]
Chassis configs will require a platform specific base bcm chassis map with deep knowledge about the lane mappings. This can be solved in a later PR, which adds the actual platform.

**Pull request merged**